src/utils.rs

```diff
@@ -32,10 +50,19 @@ pub fn retrieve_infos(
     // in various `.ledger.<field_name>` section of the binary.
     // For now we only need the API_LEVEL
     for section in elf.section_headers.iter() {
-        if let Some(Ok("ledger.api_level")) =
+        if let Some(Ok(name)) =
             elf.shdr_strtab.get(section.sh_name)
         {
-            infos.api_level = get_string_from_offset(&buffer, &(section.sh_offset as usize));
+            if matches!(name, "ledger.api_level" | ".ledger.api_level") {
+                if contains_number_representation(&buffer, &(section.sh_offset as usize)) {
+                    infos.api_level = get_string_from_offset(&buffer, &(section.sh_offset as usize));
+                }
+                else {
+                    // Assume the api_level is stored as a u8
+                    infos.api_level = buffer[section.sh_offset as usize].to_string();
+                }
+                break;
+            }
```

Comment:
Maybe you could simplify by considering/
- .ledger.api_level = int
- ledger.apli_level = string
?

Reply:
I find it best not to assume anything about the file we parse ^^

Reply:
@xchapron-ledger I have simplified it as you suggested.